clients/go/cmd/zbctl/main_test.go

```diff
@@ -209,6 +209,22 @@ var tests = []testCase{
 		cmd:        strings.Fields("--insecure delete resource 2251799813685257"),
 		goldenFile: "testdata/delete_resource.golden",
 	},
+	{
+		name:       "update unknown job timeout",
+		setupCmds:  [][]string{},
+		cmd:        strings.Fields("--insecure update timeout 2251799813685253 --timeout 10000"),
+		goldenFile: "testdata/update_unknown_job_timeout.golden",
+	},
+	{
+		name: "update job timeout",
+		setupCmds: [][]string{
+			strings.Fields("--insecure deploy resource testdata/job_model.bpmn"),
+			strings.Fields("--insecure create instance jobProcess"),
+			strings.Fields("--insecure activate jobs jobType --maxJobsToActivate 1"),
+		},
+		cmd:        strings.Fields("--insecure update timeout 2251799813685360 --timeout 10000"),
+		goldenFile: "testdata/update_job_timeout.golden",
```

Comment:
why does it work now ? What is different from the test case during our pairing session ?

Reply:
This is what I've done:
I've run `strings.Fields("--insecure activate jobs jobType --maxJobsToActivate 1")` command in my `update job timeout` test and checked what job key was generated for my particular test. I did it several times and each time the key for the job was `2251799813685360 `.

Example:
```
{
	name: "update job timeout",
	setupCmds: [][]string{
		strings.Fields("--insecure deploy resource testdata/job_model.bpmn"),
		strings.Fields("--insecure create instance jobProcess"),
	},
        cmd: strings.Fields("--insecure activate jobs jobType --maxJobsToActivate 1"),
	goldenFile: "testdata/update_job_timeout.golden",
	},
```

It seems to me that when the tests are running in sequence the same key is generated for each test every time you run it. Not exactly sure why, but this is the case.

Reply:
> I've run strings.Fields("--insecure activate jobs jobType --maxJobsToActivate 1") command in my update job timeout test and checked what job key was generated for my particular test

Smart ! I would advise providing this information as a comment

> It seems to me that when the tests are running in sequence the same key is generated for each test every time you run it

probably zeebe is using random number generator with a fixed random seed in the test environment .